packages/process/src/node/terminal-process.ts

```diff
@@ -103,8 +103,14 @@ export class TerminalProcess extends Process {
                 // be ignored.
                 if (signal === undefined || signal === 0) {
                     this.emitOnExit(code, undefined);
+                    this.onProcessExited();
+                    // Unregister process only if terminal already attached, Fix issue #2961.
+                    if (this.attached) {
+                        this.unregisterProcess();
+                    }
                 } else {
                     this.emitOnExit(undefined, signame(signal));
+                    this.unregisterProcess();
```

Comment:
I'm a bit confused about the conditionals here. Under what circumstances will we get into this `else`, and why do we want to `.unregisterProcess` immediately if we do get into this block rather than waiting for the frontend to try to attach?

Reply:
`code` <number> The exit code if the child exited on its own.
`signal` <string> The signal by which the child process was terminated.
The `'exit'` event is emitted after the child process ends. If the process exited, `code` is the final exit code of the process, otherwise `null`. If the process terminated due to receipt of a signal, `signal` is the string name of the signal, otherwise `null`. One of the two will always be non-null. (source [node.js docs](https://nodejs.org/docs/latest-v10.x/api/child_process.html#child_process_event_exit)

We will get into this `else` when child process was terminated and didn't exit on its own..

Reply:
I thought it will never happen in this case of short lived task, But after reconsideration you could be right, Should I add the check for attached there too?

Reply:
Updated..